### PR TITLE
Struct-Based route Registration

### LINKED ISSUE #71: Struct-Based Operations
**Is your feature request related to a problem? Please describe.**

I'd like to be able to define my operations via structs, instead of calling additional methods.

**Describe the solution you'd like**

Example using Fuego today:

```go
...

  server := fuego.NewServer()

  apiV1 := fuego.Group(server, "/v1")

  myReq := fuego.Get(apiV1, "/fancy-request", fancyReqHandler)
  myReq.QueryParam("item-name", "Name of the item you are working with")

...
```

Something akin to what I'd like to see (but am open to suggestions):

```go
...

  server := fuego.NewServer()

  apiV1 := fuego.Group(server, "/v1")

  myReq := fuego.Get(apiV1, fuego.Request{
      Name: "Fancy Request",
      Path: "/fancy-request",
      ID: "fancy-request-v1",
      QueryParams: []fuego.QueryParam{
        {
          Name: "item-name",
          Description: "Name of the item you are working with"
        },
      },
      Handler: fancyReqHandler,
      Tags: []string{"fancy"},
    }
  )

...
```

**Describe alternatives you've considered**

In this context, I've not yet thought of an alternative.

**Additional context**

I'll be honest, I had played w/ Huma's V2 before playing w/ Fuego :). I really appreciated being able to define structs for something like this. Felt much cleaner than calling methods over and over again.

I'll also note, that while Huma does allow using structs to define operations, and I enjoyed it, I was honestly just glad to find that Huma did it, because it felt closer to the pattern in another project that I really enjoyed, which is the [urfave/cli](https://cli.urfave.org/v2/examples/subcommands/). I've linked specifically to their `subcommands` and I think it shows a bit more of an example of where something like this _could_ go when it comes to defining routes/operations. In the context of a REST API application, consider Groups/Routes instead of Commands/Subcommands.

### PR DESCRIPTION
**Pull request by EwenQuim**
closes #71 

All fields (ID...) are not currently implemented

### CODE REVIEW

**Comment by dylanhitt** [comment]
Will take a look tomorrow.

**Comment by EwenQuim** [comment]
Thanks!

The PR is incomplete but the end result will look 80% like this.

Have a nice week-end too :)

**Comment by EwenQuim** [comment]
@dylanhitt ready for review if you have some time

File: examples/full-app-gourmet/server/server.go [hunk 1]
```diff
@@ -36,7 +36,11 @@ func (rs Ressources) Setup(
 	// With fuego, you can use any existing middleware that relies on `net/http`, or create your own
 	fuego.Use(app, chiMiddleware.Compress(5, "text/html", "text/css", "application/json"))
 
-	fuego.Handle(app, "/static/", http.StripPrefix("/static", static.Handler()), func(h http.Handler) http.Handler {
+	fuego.Register(app, fuego.Route[any, any]{
```
Comment (dylanhitt) [inline examples/full-app-gourmet/server/server.go:39]: Do you mind extending this to show off the export of `Operation`?

**Review by dylanhitt — COMMENTED** [review]
Looks good. Just that one comment. Feel free to resolve if you find it unnecessary.

**Pull request merged**